subsys/net/lib/config/init_clock_sntp.c

```diff
@@ -48,7 +59,11 @@ int net_init_clock_via_sntp(void)
 		goto end;
 	}
 
-	tspec.tv_sec = ts.seconds;
+#if CONFIG_NET_CONFIG_SNTP_TIMEZONE
+  tspec.tv_sec = ts.seconds - (UTC_0 - global_tz) * 3600;
```

Comment:
There is now some misunderstanding how timezones are used. So typically the internal clock is kept in UTC_0 but when presenting the time to the user, the timezone is taken into account and tweaked accordingly. This way for example DST handling is very easy as the internal clock does not need any tweaking when that happens.
So with this in mind, this PR is doing things that are not really needed.